src/backend/distributed/sql/udfs/get_missing_time_partition_ranges/latest.sql

```diff
@@ -0,0 +1,264 @@
+        ELSIF table_partition_interval < INTERVAL '1 week' THEN
+            current_range_from_value = date_trunc('day', start_from);
+        ELSIF table_partition_interval < INTERVAL '1 month' THEN
+            current_range_from_value = date_trunc('week', start_from);
+        ELSIF table_partition_interval = INTERVAL '3 months' THEN
+            current_range_from_value = date_trunc('quarter', start_from);
+        ELSIF table_partition_interval < INTERVAL '1 year' THEN
+            current_range_from_value = date_trunc('month', start_from);
+        ELSE
+            current_range_from_value = date_trunc('year', start_from);
+        END IF;
+
+        current_range_to_value := current_range_from_value + table_partition_interval;
+
+    ELSIF distinct_partition_interval_count = 1 THEN
+        SELECT
+        DISTINCT to_value::timestamptz - from_value::timestamptz
+        INTO table_partition_interval
+        FROM time_partitions
+        WHERE parent_table = table_name AND to_value <> '' AND from_value <> '';
+
+        IF partition_interval IS NOT NULL AND partition_interval <> table_partition_interval THEN
+            RAISE 'partition_interval does not match existing partitions'' interval';
+        END IF;
+
+        /* use initial partition as pivot to find range for missing partitions */
+        SELECT from_value::timestamptz, to_value::timestamptz
+        INTO current_range_from_value, current_range_to_value
+        FROM pg_catalog.time_partitions
+        WHERE parent_table = table_name
+        ORDER BY from_value::timestamptz ASC
+        LIMIT 1;
+
+        /* if start_from is newer than pivot's from value, go forward, else go backward */
+        IF start_from >= current_range_from_value THEN
+            WHILE current_range_from_value < start_from LOOP
+                    current_range_from_value := current_range_from_value + table_partition_interval;
+            END LOOP;
+            current_range_to_value := current_range_from_value + table_partition_interval;
+        ELSE
+            WHILE current_range_from_value > start_from LOOP
+                    current_range_from_value := current_range_from_value - table_partition_interval;
+            END LOOP;
+            current_range_to_value := current_range_from_value + table_partition_interval;
+        END IF;
+
+    ELSIF distinct_partition_interval_count > 1 THEN
+        RAISE 'each partition must cover same interval to use that function'; --TODO: Check the message
+    END IF;
+
+    /* get max_table_name_length to use it while finding partitions' name */
+    SELECT max_val
+    INTO max_table_name_length
+    FROM pg_settings
+    WHERE name = 'max_identifier_length';
+
+    /* get table_name from pg_class to generate name without schema */
+    SELECT relname
+    INTO table_name_text
+    FROM pg_class AS p JOIN pg_namespace AS n ON p.relnamespace = n.oid
+    WHERE p.oid = table_name::oid;
```

Comment:
the join with pg_namespace does not seem useful.

minor / optional: we could consider doing this in the initial query on pg_partitioned_table by joining with pg_class